cmip6_downscaling/methods/maca/core.py

```diff
@@ -251,12 +350,24 @@ def maca_construct_analogs(
     for i in range(len(y)):
         # get data from the GCM day being downscaled
         yi = y.isel(ndays_in_gcm=i)
+        print('yi', yi.load())
         # get data from the coarsened obs analogs
         ind = inds.isel(ndays_in_gcm=i).values
+        print('ind', ind)
+
         xi = X.isel(ndays_in_obs=ind).transpose('pixel_coarse', 'ndays_in_obs')
+        print('xi', xi.load())
 
         # fit model
-        lr_model.fit(xi, yi)
+
+        x_valid_mask = ~np.isnan(xi)
+        y_valid_mask = ~np.isnan(yi)
+        np.testing.assert_equal(x_valid_mask, y_valid_mask)
+
+        lr_model.fit(
+            xi[x_valid_mask],
+            yi[y_valid_mask]
+        )
```

Comment:
this is where I'm leaving off for the weekend...

The issue I was running into was that there were some nans included in the `xi` and `yi` variables and this broke the linear model fit. I was testing whether filtering nans out before bit solved the issue.